File: Proyecto-final-de-grado-main/PescaCantabriaApp/pesca_cantabria/lib/models/zona.dart
class Zona {
  int id;
  String nombre;
  String tipo;
  double latitud;
  double longitud;
  String? imageurl;

  Zona({
    required this.id,
    required this.nombre,
    required this.tipo,
    required this.latitud,
    required this.longitud,
    required this.imageurl,
  });

  factory Zona.fromJson(Map<String, dynamic> json) {
    return Zona(
      id: json['id'] as int,
      nombre: json['nombre'],
      tipo: json['tipo'],
      latitud: json['latitud'] as double,
      longitud: json['longitud'] as double,
      imageurl: json['imageurl'],
    );
  }

  @override
  String toString() {
    return "Zona{id: $id, nombre: $nombre, tipo: $tipo, coordenadas: $latitud $longitud, imageurl: $imageurl}";
  }
}

File: Proyecto-final-de-grado-main/GestorAPI/src/main/java/TFG/model/Zona.java
package TFG.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Entity
@Table(name = "zona")
public class Zona {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idzona", nullable = false)
    private Integer id;

    @NotNull
    @Lob
    @Column(name = "tipo", nullable = false)
    private String tipo;

    @Size(max = 45)
    @NotNull
    @Column(name = "nombre", nullable = false, length = 45)
    private String nombre;

    @Column(name = "latitud")
    private Double latitud;

    @Column(name = "longitud")
    private Double longitud;

    @Size(max = 200)
    @Column(name = "imageurl", length = 200)
    private String imageurl;

    public String getImageurl() {
        return imageurl;
    }

    public void setImageurl(String imageurl) {
        this.imageurl = imageurl;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public Double getLatitud() {
        return latitud;
    }

    public void setLatitud(Double latitud) {
        this.latitud = latitud;
    }

    public Double getLongitud() {
        return longitud;
    }

    public void setLongitud(Double longitud) {
        this.longitud = longitud;
    }
}